src/vivilux/layers.py

```diff
@@ -174,6 +176,24 @@ def RunProcesses(self):
     def AddProcess(self, process: Process):
         process.AttachLayer(self)
 
+    def AddMonitor(self, monitor: Monitor):
+        self.monitors.append(monitor)
```

Comment:
This was similar to my original implementation, but should we switch from a list to a dict (or both data structures is dict iterates faster)? That way we can turn on and off specific monitors by name.

Reply:
That makes sense, I will store monitors in a dict instead.